test/e2e/operator_test.go

```diff
@@ -217,12 +221,12 @@ func TestExternalDNSWithRoute(t *testing.T) {
 
 		// verify dns records has been created for the route host.
 		if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
-			cname, err := lookupCNAMEMiekg(testRouteHost, nameSrv)
+			cname, err := lookupCNAME(testRouteHost, nameSrv)
 			if err != nil {
 				t.Logf("Waiting for DNS record: %s, error: %v", testRouteHost, err)
 				return false, nil
 			}
-			if equalFQDN(cname, targetRouterCName) {
+			if strings.Contains(cname, targetRouterCName) {
```

Comment:
This should be reverted, we want the exact match, not substring.

Reply:
ok